includes/Core/Authentication/Setup_V2.php

```diff
@@ -54,15 +54,23 @@ public function handle_action_setup_start() {
 			wp_die( esc_html__( 'Site Kit is not configured to use the authentication proxy.', 'google-site-kit' ) );
 		}
 
-		if ( $this->credentials->has() ) {
-			$this->google_proxy->sync_site_fields( $this->credentials, 'sync' );
+		$required_scopes = $this->authentication->get_oauth_client()->get_required_scopes();
+		$this->google_proxy->with_scopes( $required_scopes );
+
+		$oauth_setup_redirect = $this->credentials->has()
+			? $this->google_proxy->sync_site_fields( $this->credentials, 'sync' )
+			: $this->google_proxy->register_site();
+
+		if ( is_wp_error( $oauth_setup_redirect ) || ! filter_var( $oauth_setup_redirect, FILTER_VALIDATE_URL ) ) {
+			wp_die( esc_html__( 'The request to the authentication proxy has failed. Please, try again later.', 'google-site-kit' ) );
 		}
 
 		if ( $redirect_url ) {
 			$this->user_options->set( OAuth_Client::OPTION_REDIRECT_URL, $redirect_url );
 		}
 
-		$this->redirect_to_proxy();
+		wp_redirect( $oauth_setup_redirect ); // phpcs:ignore WordPress.Security.SafeRedirect.wp_redirect_wp_redirect
```

Comment:
Can't we use `wp_safe_redirect` here?

Reply:
No, because the redirect URL has a different domain.

Reply:
I don't know because I can't verify it on my local. When I try to set up the plugin with the V2 setup flow, I get the following error: 

```
We were unable to send credentials to your WordPress site. Please go back to your site and restart the setup. The following error was returned: No route was found matching the URL and request method.
```

Is it something wrong on the WP side or you need to adjust something on the proxy side?

Reply:
I just tested this (note that you have to use the staging proxy) and found a few other small, but crucial problems that were preventing the flow from working. But yes, I then verified that `wp_safe_redirect` works here as expected, since we allowlist the host.